maat-court-data-api/src/test/java/gov/uk/courtdata/eform/controller/EformsDecisionHistoryControllerTest.java

```diff
@@ -0,0 +1,147 @@
+                .andExpect(jsonPath("$.iojResult").value("Pass"))
+                .andExpect(jsonPath("$.repDecision").value("Granted"))
+                .andExpect(jsonPath("$.wroteToResults").value("Y"));
+
+        verify(eformsDecisionHistoryService, times(1)).getPreviousEformsDecisionHistoryRecordWroteToResult(12345);
+    }
+
+    @Test
+    void shouldSuccessfullyDeleteEformDecisionHistoryForGivenUSN() throws Exception {
+        mvc.perform(MockMvcRequestBuilders.delete(BASE_ENDPOINT_FORMAT+ "/" +12345)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+
+        verify(eformsDecisionHistoryService, times(1)).deleteEformsDecisionHistory(12345);
+    }
+
+    @Test
+    void shouldSuccessfullyUpdateEformDecisionHistory() throws Exception {
+        EformsDecisionHistory eformsDecisionHistory = EformsDecisionHistory.builder().id(3).usn(12345).repId(45635).fundingDecision("Granted").iojResult("Pass").build();
+        String requestJson = "{\"wroteToResults\":\"Y\"}";
+        Map<String,Object> updateFields = new ObjectMapper().readValue(requestJson, HashMap.class);
+        mvc.perform(MockMvcRequestBuilders.patch(BASE_ENDPOINT_FORMAT+ "/" +12345).content(requestJson)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        verify(eformsDecisionHistoryService, times(1)).updateEformsDecisionHistoryFields(12345, updateFields);
+    }
+
+    @Test
+    void shouldSuccessfullyCreateEformDecisionHistory() throws Exception {
+        EformsDecisionHistory eformsDecisionHistory = EformsDecisionHistory.builder().id(1).usn(123).repId(45635).fundingDecision("Granted").iojResult("Pass").build();
+        mvc.perform(MockMvcRequestBuilders.post(BASE_ENDPOINT_FORMAT).content(eformsDecisionHistory())
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        verify(eformsDecisionHistoryService, times(1)).createEformsDecisionHistory(eformsDecisionHistory);
+    }
+    @Test
+    void shouldSuccessfullyGetAllEformDecisionHistoryForGivenUSN() throws Exception {
+        EformsDecisionHistory eformsDecisionHistory = EformsDecisionHistory.builder().id(1).usn(123).repId(45635).fundingDecision("Granted").iojResult("Pass").build();
+        List<EformsDecisionHistory> eformsDecisionHistoryList = List.of(eformsDecisionHistory);
+        when(eformsDecisionHistoryService.getAllEformsDecisionHistory(123))
+                .thenReturn(eformsDecisionHistoryList);
+
+        mvc.perform(MockMvcRequestBuilders.get(BASE_ENDPOINT_FORMAT+ "/" +123)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().json("["+eformsDecisionHistory()+"]"));
+        verify(eformsDecisionHistoryService, times(1)).getAllEformsDecisionHistory(123);
+    }
+
+    private String eformsDecisionHistory(){
```

Comment:
I would consider storing this json as a .json file in `test/resources `and using `FileUtils.readResourceToString()` to read it.

> laa-maat-court-data-api/maat-court-data-api/src/test/java/gov/uk/courtdata/testutils/FileUtils.java

Reply:
As discussed, we are OK with validating the data based on fields instead of adding new files